WooCommerce/Classes/ViewRelated/WhatsNew/WhatsNewHostingController.swift

```diff
@@ -0,0 +1,28 @@
+import SwiftUI
+
+/// WhatsNewHostingController wrapper to handle all the specific presentation details and trait handling
+final class WhatsNewHostingController: UIHostingController<ReportList> {
+    override init(rootView: ReportList) {
+        super.init(rootView: rootView)
+        if UIDevice.isPad() {
+            preferredContentSize = Layout.iPadContentSize
+        }
+        modalPresentationStyle = .formSheet
+    }
+
+    override var traitCollection: UITraitCollection {
+        self.presentingViewController?.traitCollection ?? .current
+    }
```

Comment:
Yes it is! Once you define a `preferredContentSize` for that view (iPad only), the traits are now `.compact` due to the view size. This makes it impossible to distinguish what is iPhone and what is iPad to apply the right sizing, margins, etc.

Reply:
Oh! The I think it's worth to add a comment explain why it is necessary, or we risk that a future maintainer removes it because they think it is not needed.

Reply:
Makes total sense. I've added the docs!
Done! ✅